src/ui/public/courier/fetch/strategy/__tests__/search.js

```diff
@@ -31,6 +31,57 @@ describe('ui/courier/fetch/strategy/search', () => {
     });
   });
 
+  describe('#handleResponseError()', () => {
+    let error;
+    beforeEach(() => {
+      error = { status: 404, body: { error: { index: '[-*]' } } };
+    });
+
+    it('recovers 404 for index -* with empty response', () => {
+      let resp;
+      search.handleResponseError(reqsFetchParams, error).then(val => resp = val);
+      $rootScope.$apply();
+
+      expect(resp.responses).not.to.be(undefined);
+    });
+
+    it('mocks all of the bundled searches', () => {
+      let resp;
+      reqsFetchParams.push({});
+      search.handleResponseError(reqsFetchParams, error).then(val => resp = val);
+      $rootScope.$apply();
+
+      expect(Array.isArray(resp.responses)).to.be(true);
+      expect(resp.responses.length).to.be(2);
+      resp.responses.forEach(res => {
+        expect(res.hits.total).to.be(0);
+        expect(res.hits.hits.length).to.be(0);
+      });
+    });
+
+    context('when not a 404', () => {
```

Comment:
Didn't we decide on not using `context` and just sticking with nested `describe`s?